### PR TITLE
OMR work addition to 2.3.2

### PR DESCRIPTION
**Pull request by nasehim7**
Porting the changes from imeta to master. Changes made wherever was required to make it compatible. Major Update: Edata.cpp addition which does writing of the metadata for Audiveris. Moreover, Adding Grace Notes, Bracket, Repeat Dots, Bracketed Time Signature and Tuplet Implementation.

### CODE REVIEW

**Comment by nasehim7** [comment]
Not building @lasconic :/ Strange! Works like a charm on my system.

**Comment by Jojo-Schmitz** [comment]
Replacement for #3782

**Comment by lasconic** [comment]
This PR is for branch 2.3.2

**Comment by Jojo-Schmitz** [comment]
Yes, but it needs a Qt newer than 5.4 for GreyScale (?)

**Comment by nasehim7** [comment]
So how to go about it?

File: build/appveyor/before_build.bat [hunk 1]
```diff
@@ -7,8 +7,8 @@ SET OLD_PATH=%PATH%
 :: copy dependencies
 START " " /wait "7z" x -y dependencies.7z > nul
 CD dependencies
-XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw491_32\i686-w64-mingw32" /E /Y
-XCOPY lib "C:\Qt\Tools\mingw491_32\lib" /E /Y
+XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw53_32\i686-w64-mingw32" /E /Y
```
Comment (Jojo-Schmitz) [inline build/appveyor/before_build.bat]: mingw530_32 here

File: build/appveyor/before_build.bat [hunk 2]
```diff
@@ -7,8 +7,8 @@ SET OLD_PATH=%PATH%
 :: copy dependencies
 START " " /wait "7z" x -y dependencies.7z > nul
 CD dependencies
-XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw491_32\i686-w64-mingw32" /E /Y
-XCOPY lib "C:\Qt\Tools\mingw491_32\lib" /E /Y
+XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw53_32\i686-w64-mingw32" /E /Y
+XCOPY lib "C:\Qt\Tools\mingw53_32\lib" /E /Y
```
Comment (Jojo-Schmitz) [inline build/appveyor/before_build.bat]: but mingw53_32 here

File: build/mingw32.mingw.cmake [hunk 3]
```diff
@@ -4,8 +4,8 @@
 
 set (CMAKE_SYSTEM_NAME "Windows")
 
-set (CROSS C:/Qt/Tools/mingw491_32)
-set (CROSSQT C:/Qt/5.4/mingw491_32)
+set (CROSS C:/Qt/Tools/mingw53_32)
```
Comment (Jojo-Schmitz) [inline build/mingw32.mingw.cmake]: see above

Comment (Jojo-Schmitz) [reply build/mingw32.mingw.cmake]: it doesn't seem too logical, but yes, mingw530_32 here and mingw53_32 there

Comment (anatoly-os) [reply build/mingw32.mingw.cmake]: It is like 5.3.0 and 5.3 version.
I believe it is due to the agile approach. It leads to such small inconsistencies and less quality, but increases speed of development.

File: build/appveyor/before_build.bat (hunk 2 above)
Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: Sorry, `C:\Qt\Tools\mingw530_32\lib` here too

File: build/appveyor/before_build.bat [hunk 4]
```diff
@@ -22,12 +22,12 @@ python build/add-mc-keys.py %MC_CONSUMER_KEY% %MC_CONSUMER_SECRET%
...
 
 :: get revision number
-SET PATH=C:\Qt\5.4\mingw491_32\bin;C:\Qt\Tools\mingw491_32\bin;%PATH%
+SET PATH=C:\Qt\5.9.6\mingw53_32\bin;%PATH%
 mingw32-make -f Makefile.mingw revision
 SET /p MSversion=<mscore\revision.h
 
 :: CMake refuses to generate MinGW Makefiles if sh.exe is in the PATH (C:\Program Files\Git\usr\bin)
-SET PATH=C:\Qt\5.4\mingw491_32\bin;C:\Qt\Tools\mingw491_32\bin;C:\Program Files (x86)\CMake\bin;C:\Program Files\7-Zip;C:\ccache\bin;C:\Tools\curl\bin;%WIX%\bin;C:\Windows\system32;C:\Windows
+SET PATH=C:\Qt\5.9.6\mingw53_32\bin;C:\Qt\Tools\mingw53_32\bin;C:\Program Files (x86)\CMake\bin;C:\Program Files\7-Zip;C:\ccache\bin;C:\Tools\curl\bin;%WIX%\bin;C:\Windows\system32;C:\Windows
```
Comment (Jojo-Schmitz) [inline build/appveyor/before_build.bat]: `C:\Qt\Tools\mingw53_32\lib here too

File: build/appveyor/before_build.bat [hunk 5]
```diff
@@ -22,12 +22,12 @@ python build/add-mc-keys.py %MC_CONSUMER_KEY% %MC_CONSUMER_SECRET%
 )
 
 :: get revision number
-SET PATH=C:\Qt\5.4\mingw491_32\bin;C:\Qt\Tools\mingw491_32\bin;%PATH%
+SET PATH=C:\Qt\5.9.6\mingw53_32\bin;%PATH%
```
Comment (Jojo-Schmitz) [inline build/appveyor/before_build.bat]: `SET PATH=C:\Qt\5.9\mingw53_32\bin;C:\Qt\Tools\mingw530_32\bin;%PATH%`

Comment (nasehim7) [reply build/appveyor/before_build.bat]: 5.9.6, right?

File: build/appveyor/before_build.bat (hunk 4 above)
Comment (nasehim7) [reply build/appveyor/before_build.bat]: Should I be adding C:\Qt\Tools\mingw53_32\lib to PATH or editing C:\Qt\Tools\mingw53_32\bin; to C:\Qt\Tools\mingw53_32\lib;?

File: build/appveyor/before_build.bat (hunk 5 above)
Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: Grr, yes, of course

File: build/appveyor/before_build.bat (hunk 4 above)
Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: Oh boy, fat fingers again, sorry, I meant `C:\Qt\Tools\mingw53_32\bin`

Comment (nasehim7) [reply build/appveyor/before_build.bat]: No problem :D

Comment (nasehim7) [reply build/appveyor/before_build.bat]: feels like we have no change to be done here

Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: Unfortunatly we do, as I screwed up again,  I meant `C:\Qt\Tools\mingw530_32\bin`

Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: It is just way to warm here

File: build/appveyor/before_build.bat [hunk 6]
```diff
@@ -7,8 +7,8 @@ SET OLD_PATH=%PATH%
 :: copy dependencies
 START " " /wait "7z" x -y dependencies.7z > nul
 CD dependencies
-XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw491_32\i686-w64-mingw32" /E /Y
-XCOPY lib "C:\Qt\Tools\mingw491_32\lib" /E /Y
+XCOPY i686-w64-mingw32 "C:\Qt\Tools\mingw530_32\i686-w64-mingw32" /E /Y
+XCOPY lib "C:\Qt\Tools\mingw53_32\lib" /E /Y
```
Comment (Jojo-Schmitz) [inline build/appveyor/before_build.bat]: OK, again, here we need `XCOPY lib "C:\Qt\Tools\mingw530_32\lib" /E /Y`

Comment (nasehim7) [reply build/appveyor/before_build.bat]: Aaahh! My bad. Sorted. :D

**Comment by Jojo-Schmitz** [comment]
Next problem: https://ci.appveyor.com/project/MuseScore/musescore/build/1.0.2216#L123
"This code requites Qt 4.x" from build/FindQt5Transitional.cmake line 88, `find_package(QT4 ${QT_MIN_VERSION} REQUIRED ${_components})`

**Comment by nasehim7** [comment]
@Jojo-Schmitz Firstly it tries to look for QT5, right?

File: build/appveyor/before_build.bat (hunk 6 above)
Comment (Jojo-Schmitz) [reply build/appveyor/before_build.bat]: Adding a `/I` to the `XCOPY` commands should prevent the hang

**Comment by Jojo-Schmitz** [comment]
Not sure, I don't get the logic to runs thru

**Comment by nasehim7** [comment]
I think we will have to make some edits there too. I will check all of that. :D

**Comment by Jojo-Schmitz** [comment]
The Travis fail for the Mac build seems unrelated, IIRC for PRs it doesn't fully run anyway

**Comment by nasehim7** [comment]
Will have to check for the reason. Config issue it seems. I have just
imported 5.9.2 in my path and no code changes as such. Closer look will get
it resolved. Lets see. :)

**Comment by nasehim7** [comment]
@Jojo-Schmitz except for the Mac Build, all crashes report about Format_GrayScale8.

**Comment by lasconic** [comment]
Then Qt 5.9 is not used on Appveyor yet.

**Comment by nasehim7** [comment]
Sad part is Travis is failing too for Linux.

**Comment by lasconic** [comment]
That's normal, travis is also setup to built with 5.4 and needs to be changed. But priority is for appveyor to work. So why doesn't it used 5.9 ?

**Comment by Jojo-Schmitz** [comment]
Because of build/FindQt5Transitional.cmake, line 88, see https://ci.appveyor.com/project/MuseScore/musescore/build/1.0.2216#L123

**Comment by nasehim7** [comment]
What if we remove this line from .cmake? Does the code really require 4.x support, like anything specific? if we remove this line, we get: https://ci.appveyor.com/project/MuseScore/musescore/build/1.0.2260
was just curious so tried this.

**Comment by anatoly-os** [comment]
Since Herve is aware of the generated data set and already tested it to apply in Audiveris, we don't need to keep this PR opened. @nasehim7 thank you for your work! I believe it will significantly improve OMR algorithms to get better results.

**Pull request closed without merging**